### PR TITLE
convert jit_language_reference.rst to jit_language_reference.md

### PR DESCRIPTION
**Pull request by Rachel0619**
Part of changes https://github.com/pytorch/pytorch/issues/155023 (parent PR https://github.com/pytorch/pytorch/pull/155429)

- converted jit_language_reference.rst to jit_language_reference.md

@pytorchbot label "topic: docs"
@pytorchbot label "topic: not user facing"
@pytorchbot label docathon-h1-2025
@pytorchbot label module: docs

### CODE REVIEW

File: docs/source/jit_language_reference.md [hunk 1]
```diff
@@ -840,58 +871,57 @@ value should be treated as a constant.
...
+- `bool`
+- `torch.device`
+- `torch.layout`
+- `torch.dtype`
+- tuples containing supported types
+- `torch.nn.ModuleList` which can be used in a TorchScript for loop
 
-.. _module attributes:
+(module-attributes)=
+(Module Attributes)=
```
Comment (svekars) [inline docs/source/jit_language_reference.md:908]:
```suggestion
```

File: docs/source/jit_language_reference.md [hunk 2]
```diff
@@ -398,140 +426,137 @@ Types produced by :func:`collections.namedtuple <collections.namedtuple>` can be
...
 
     ...
 
+```
 
-.. _jit_iterables:
-
-Iterables
-^^^^^^^^^
+(jit-iterables)=
```
Comment (svekars) [inline docs/source/jit_language_reference.md]:
```suggestion
(jit_iterables)=
```

File: docs/source/jit_language_reference.md [hunk 3]
```diff
@@ -301,89 +321,97 @@ similar to how you would declare a TorchScript function:
...
+  >
+  > Will result in:
+  >
+  > ```
+  > RuntimeError:
+  > Tried to set nonexistent attribute: x. Did you forget to initialize it in __init__()?:
+  > def assign_x(self):
+  >   self.x = torch.rand(2, 3)
+  >   ~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
+  > ```
```
Comment (svekars) [inline docs/source/jit_language_reference.md:352]: Not sure why there is a leading > here. Can you explain?

File: docs/source/jit_language_reference.md (hunk 1 above)
Comment (Rachel0619) [reply docs/source/jit_language_reference.md:908]: there's a `:any:` at line 146, not sure whether deleting `(Module Attributes)=` will break that reference

File: docs/source/jit_language_reference.md (hunk 3 above)
Comment (Rachel0619) [reply docs/source/jit_language_reference.md:352]: It's for the code block to be visually aligned with the bullet point and it's rendered like this:

image

**Review by svekars — APPROVED** [review]
LGTM

File: docs/source/jit_language_reference.md [hunk 4]
```diff
@@ -23,19 +24,19 @@
         return original_trace(obj, *args, **kwargs)
 
     torch.jit.trace = trace_wrapper
+```
 
-.. _language-reference:
+(language-reference)=
```
Comment (thatgeeman) [inline docs/source/jit_language_reference.md:29]: @Rachel0619 shouldnt this be `jit_language_reference`?

based on a reference here: https://raw.githubusercontent.com/pytorch/pytorch/refs/heads/main/docs/source/jit.rst

CC: @svekars

Comment (Rachel0619) [reply docs/source/jit_language_reference.md:29]: Hi @thatgeeman I think in the below code (if that's what you are referring to), they are referencing `jit_language_reference.md` file and `jit_language_reference_v2.md` file instead of the headings.

CC: @svekars

```
.. toctree::
    :maxdepth: 1
    :caption: Language Reference
    :hidden:

    jit_language_reference


.. toctree::
    :maxdepth: 1

    jit_language_reference_v2
```

**Comment by thatgeeman** [comment]
Yes that's true, that's for the clarification (y)
